### ISSUE TITLE
Labels Break JSHint

### ISSUE THREAD

**Issue by iamlikeme-x** [jshint/jshint#3055]
The following produces the error `E053 - Export declarations are only allowed at the top level of module scope.`

Whilst I appreciate that `const` and `export` shouldn't be used in production code (unless controlled environments, eg. Electron - the use case I have) this still shouldn't be happening AFAIK.

``` javascript
testLabel: {
    break testLabel;
}
export const test = 'test';
```

It appears JSHint isn't matching the closing brace properly, as the following supresses the error despite being syntactically incorrect:

``` javascript
testLabel: {
    break testLabel;
}
}
export const test = 'test';
```

**Comment by jugglinmike** [comment]
Confirmed. Here's a further-reduced test case:

```js
// jshint esversion: 6, module: true
x: {}
export default null;
```

You're right to suspect improper scope tracking. It would appear that the same underlying bug is effecting the binding tracking mechanism. Note how `undef: true` ought to trigger a warning in the following code but currently does not:

```js
// jshint esversion: 6, module: true, undef: true
{ let y; }
void y;
```

Thanks for the report!

**Comment by jugglinmike** [comment]
Submitted a patch here: https://github.com/jshint/jshint/pull/3057

**Comment by jugglinmike** [comment]
This was resolved in gh-3057 and [subsequently released as part of JSHint version 2.9.5](https://github.com/jshint/jshint/releases/tag/2.9.5).
